fvm/blueprints/contractDeployment.go

```diff
@@ -0,0 +1,38 @@
+package blueprints
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const ContractDeploymentAuthorizedAddressesPathDomain = "storage"
+const ContractDeploymentAuthorizedAddressesPathIdentifier = "authorizedAddressesToDeployContracts"
+
+const setContractDeploymentAuthorizersTransactionTemplate = `
+transaction {
+		prepare(signer: AuthAccount) {
+			// this drops previous list
+			signer.load<[Address]>(from: /%s/%s)
+			signer.save([%s], to: /%s/%s)
+		}
+}
+`
+
+// SetContractDeploymentAuthorizersTransaction returns a transaction for setting storage path ...
+func SetContractDeploymentAuthorizersTransaction(serviceAccount flow.Address, authorized []flow.Address) *flow.TransactionBody {
```

Comment:
Ideally the transaction would call a contract function that does the update (load + save), but also emits an event. 

I guess it's a chicken-and-egg problem though, because to deploy such a contract, the authorization check is called, which reads the stored addresses, which don't exist yet.

However, it would still be nice to use a contract function for subsequent updates.

Reply:
I guess this will be used very rarely (probably only once). so I didn't optimize the logic here

Reply:
I think we should leave it the way it is, anyway mid term we should remove all this logic and open it up to all accounts.